### PR TITLE
WIP: edition 2018

### PR DESCRIPTION
**Pull request by burrbull**
Make crate compatible with Rust 1.31 beta and "Edition 2018"

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -46,23 +49,21 @@ version = "0.1.3"
...
 [dev-dependencies.either]
 default-features = false
-version = "1.4.0"
+version = "1.5.0"
 
 [dev-dependencies.jnet]
-git = "https://github.com/japaric/jnet"
-rev = "df96b408049ca952ad7844d6552e87cf8fc18d2a"
+#git = "https://github.com/japaric/jnet"
+git = "https://github.com/burrbull/jnet"
```
Comment (therealprof) [inline Cargo.toml]: Not happy about this and the line above. Can we crate jnet?

**Comment by burrbull** [comment]
Waiting for #https://github.com/japaric/stm32f103xx/pull/27 to be merged.

File: Cargo.toml (hunk 1 above)
Comment (TeXitoi) [reply Cargo.toml]: The examples using this crate does not really have any reason to be in this crate. What about removing all the examples that rely on not published on crates.io crates?

Comment (TeXitoi) [reply Cargo.toml]: @therealprof OK

**Comment by burrbull** [comment]
update stm32f103xx to 0.11-alpha.1
revert jnet version
@therealprof

File: Cargo.toml [hunk 2]
```diff
@@ -46,23 +51,21 @@ version = "0.1.3"
...
 rev = "df96b408049ca952ad7844d6552e87cf8fc18d2a"
 
 [dev-dependencies.motor-driver]
 git = "https://github.com/japaric/motor-driver"
-rev = "00c8b15223643090d69f1acfb8b7a7a43a440417"
 
 [dev-dependencies.mpu9250]
-git = "https://github.com/japaric/mpu9250"
-rev = "8f9ecad690093cb71c41301ca5e5705706150610"
+version = "0.7.3"
```
Comment (TeXitoi) [inline Cargo.toml]: you can put it in the dev-dependencies section

File: Cargo.toml [hunk 3]
```diff
@@ -72,16 +75,17 @@ version = "1.0.33"
 version = "1.0.33"
 
 [dev-dependencies.serde-json-core]
-git = "https://github.com/japaric/serde-json-core"
-rev = "6f12b77c1ffeae167989fe06e0d8b15978bd6d18"
+version = "0.0.1"
```
Comment (TeXitoi) [inline Cargo.toml]: you can put it in the dev-dependencies section

File: examples/enc28j60-reactive.rs.outdated [hunk 4]
```diff
@@ -2,33 +2,33 @@
 
 #![deny(unsafe_code)]
 #![deny(warnings)]
-#![feature(nll)]
-#![feature(proc_macro)]
+#![no_main]
```
Comment (TeXitoi) [inline examples/enc28j60-reactive.rs.outdated]: If this file compile, please remove the outdated at the end of the file

File: examples/mpu9250-kalman-pitch.rs.outdated [hunk 5]
```diff
@@ -2,38 +2,32 @@
 
 #![deny(unsafe_code)]
 #![deny(warnings)]
-#![feature(proc_macro)]
+#![no_main]
```
Comment (TeXitoi) [inline examples/mpu9250-kalman-pitch.rs.outdated]: same for this one

File: examples/mpu9250-log.rs.outdated [hunk 6]
```diff
@@ -2,33 +2,31 @@
 
 #![deny(unsafe_code)]
 #![deny(warnings)]
-#![feature(proc_macro)]
 #![no_std]
+#![no_main]
```
Comment (TeXitoi) [inline examples/mpu9250-log.rs.outdated]: Same here

File: examples/reactive-blinky.rs.outdated [hunk 7]
```diff
@@ -1,61 +1,47 @@
 //! Blinks an LED
 
-#![feature(proc_macro)]
 #![deny(unsafe_code)]
 #![deny(warnings)]
+#![no_main]
```
Comment (TeXitoi) [inline examples/reactive-blinky.rs.outdated]: same here

File: examples/reactive-serial-circ.rs [hunk 8]
```diff
@@ -0,0 +1,76 @@
+//! Blinks an LED
```
Comment (TeXitoi) [inline examples/reactive-serial-circ.rs]: I don't see any blinking led in this UART DMA demo ;-)

File: Cargo.toml [hunk 9]
```diff
@@ -5,14 +5,17 @@ description = "HAL for the STM32F100xx family of microcontrollers"
...
 
 [dependencies]
 cortex-m = "0.5.7"
 nb = "0.1.1"
-stm32f103xx = "0.10.0"
-cortex-m-rt = "0.6.3"
+cortex-m-rt = "0.6.5"
+
+[dependencies.stm32f103xx]
+version = "=0.11.0-alpha.1"
```
[the remark this answers is not in the document]
Comment (burrbull) [reply Cargo.toml]: rtfm 0.4 require device crate to be generated with svd2rust v0.14

File: examples/reactive-serial-circ.rs (hunk 8 above)
Comment (burrbull) [reply examples/reactive-serial-circ.rs]: green_bl

File: Cargo.toml (hunk 9 above)
[the remark this answers is not in the document]
Comment (TeXitoi) [reply Cargo.toml]: 0.11 is not? :cry:

[the remark this answers is not in the document]
Comment (TeXitoi) [reply Cargo.toml]: I think there is a misunderstanding, I don't mean switch back to 0.10, I mean using the dependencies section.

File: examples/reactive-blinky.rs.outdated (hunk 7 above)
Comment (burrbull) [reply examples/reactive-blinky.rs.outdated]: This file can't be compiled. SysTick exception is missing. I don't know why.
https://github.com/japaric/cortex-m-rtfm/issues/101

File: examples/mpu9250-kalman-pitch.rs.outdated (hunk 5 above)
Comment (burrbull) [reply examples/mpu9250-kalman-pitch.rs.outdated]: Imu struct mpu9250 crate is private, so I don't know how to create _static_ init value for rtfm.

File: examples/reactive-blinky.rs.outdated (hunk 7 above)
Comment (TeXitoi) [reply examples/reactive-blinky.rs.outdated]: Maybe it can be updated to use a normal timer?

Comment (burrbull) [reply examples/reactive-blinky.rs.outdated]: Of course, we can. But I think this is not the goal of this PR.

File: examples/mpu9250-kalman-pitch.rs.outdated (hunk 5 above)
Comment (TeXitoi) [reply examples/mpu9250-kalman-pitch.rs.outdated]: OK

**Comment by TeXitoi** [comment]
Compiling on beta \o/ https://travis-ci.org/TeXitoi/rusty-clock/jobs/464411252

I don't have the hardware here, I'll try as soon as possible.

**Comment by TeXitoi** [comment]
working on stable \o/

**Comment by TeXitoi** [comment]
stm32f103xx v0.11.0 is out, only waiting for rtfm 0.4

**Comment by TeXitoi** [comment]
I think we can merge this with the update to stm32f103xx v0.11.0. depending on the beta rtfm is OK as it is a dev dependency.

**Comment by burrbull** [comment]
Done.

**Comment by TeXitoi** [comment]
@therealprof OK to merge?

**Comment by TeXitoi** [comment]
@therealprof As you have requested changes, I let you dismiss your review or approve and merge, as you wish.

**Pull request merged**